WalletWasabi/Models/ChaumianCoinJoin/CcjRound.cs

```diff
@@ -306,6 +308,61 @@ public async Task ExecuteNextPhaseAsync(CcjRoundPhase expectedPhase)
 							.Shuffle()
 							.BuildTransaction(false);
 
+						// 8. Try optimize fees.
+						try
+						{
+							// 8.1. Estimate the current FeeRate. Note, there are no signatures yet!
+							int estimatedSigSizeBytes = UnsignedCoinJoin.Inputs.Count * Constants.P2wpkhInputSizeInBytes;
+							int estimatedFinalTxSize = UnsignedCoinJoin.GetVirtualSize() + estimatedSigSizeBytes;
```

Comment:
I think this is not okay because the virtual size of a unsigned transaction doesn't have much sense. In other words:
```
// signedTx = sign(unsignedTx)
var estimatedTxSize = unsignedTx.GetVirtualSize() + (InputCount *  P2wpkhInputSizeInBytes);
var realTxSize = signedTx.GetVirtualSize();

var goodEstimator = (1.1 * estimatedTxSize) > realTxSize && realTxSize > (estimatedTxSize * 0.9);
Console.WriteLine(goodEstimator); // False
```

Reply:
Just to show my point with code, see: https://github.com/lontivero/NBitcoinPoC/blob/master/Program.cs

Reply:
Right. Fixed.